site/app/models/DisplayImage.php

```diff
@@ -0,0 +1,212 @@
+
+    /** @prop @var string The image mime type */
+    protected $mime_type;
+
+    /**
+     * DisplayImage constructor
+     *
+     * Determines the path to the display image based on the display image state.  Also tests to ensure the image
+     * is readable at instantiation time.
+     *
+     * @param Core $core
+     * @param string $user_id
+     * @param string $display_image_state
+     * @throws BadArgumentException An illegal display_image_state was passed in
+     * @throws FileReadException Unable to read the file at the selected location for whatever reason
+     * @throws \ImagickException
+     */
+    public function __construct(Core $core, string $user_id, string $display_image_state) {
+        parent::__construct($core);
+
+        // Validate passed image state
+        if (!in_array($display_image_state, self::LEGAL_IMAGE_STATES)) {
+            throw new BadArgumentException('Unknown display_image_state!');
+        }
+
+        $sub_dir = null;
+        if ($display_image_state === 'system' || $display_image_state === 'flagged') {
+            $sub_dir = 'system_images';
+        }
+        elseif ($display_image_state === 'preferred') {
+            $sub_dir = 'user_images';
+        }
+
+        $data_dir = $this->core->getConfig()->getSubmittyPath();
+        $image_folder_dir = FileUtils::joinPaths($data_dir, 'user_data', $user_id, $sub_dir);
+
+        // If can't open image folder then return without setting $path
+        if (!is_readable($image_folder_dir)) {
+            throw new FileReadException('Unable to read from the image folder.');
+        }
+
+        // Order files by newest time stamp first
+        $files = FileUtils::getAllFilesTrimSearchPath($image_folder_dir, 0);
+        usort($files, function ($a, $b) {
+            return filemtime($a) <= filemtime($b);
+        });
+
+        // Ensure image is readable
+        if (empty($files) || !is_readable($files[0])) {
+            throw new FileReadException('Unable to read the display image.');
+        }
+
+        $imagick = new \Imagick($files[0]);
+
+        $this->mime_type = $imagick->getImageMimeType();
+        $this->display_image_state = $display_image_state;
+        $this->path = $files[0];
+    }
+
+    /**
+     * Get the display image encoded as a base64 string
+     *
+     * You may optionally resize the image.
+     *
+     * You may pass both parameters to have fine control over both image dimensions.
+     *
+     * You may pass only a single parameter and leave the other one null.
+     * Doing so will resize that dimension to the specified length, while maintaining the form factor of the image.
+     *
+     * @param int|null $cols Length in pixels to resize this dimension to
+     * @param int|null $rows Length in pixels to resize this dimension to
+     * @return string The base64 encoding of the image
+     * @throws \ImagickException
+     */
+    public function getImageBase64(int $cols = null, int $rows = null): string {
```

Comment:
```suggestion
    public function getImageBase64(?int $cols = null, ?int $rows = null): string {
```

https://www.php.net/manual/en/migration71.new-features.php#migration71.new-features.nullable-types